dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/IdentityStorePermissionCheckFactory.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.security.javaeesec.identitystore;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+/**
+ * Factory to check if "javax.security.enterprise.identitystore.IdentityStorePermission"
+ * exists, and create an implementation to check the identity store based on
+ * Security 3.0 vs 4.0+ environment.
+ */
+
+public class IdentityStorePermissionCheckFactory {
+
+    private static final TraceComponent tc = Tr.register(IdentityStorePermissionCheckFactory.class);
+    private static IdentityStorePermissionCheck instance = null;
+
+    /**
+     * Get an instance of an identity store permission checker that is valid for
+     * all Jakakata Security X.X environments.
+     *
+     * Put guards in place to ensure Singleton pattern only.
+     *
+     * @return An implementation of IdentityStorePermissionCheck
+     */
+    static synchronized IdentityStorePermissionCheck getInstance() {
+        if (instance == null) {
+            // First, check if IdentityStorePermission class exists (up to Jakarta Security 3.0)
+            try {
+                Class.forName("javax.security.enterprise.identitystore.IdentityStorePermission");
+                if (tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Using Jakarta Security 3.0 IdentityStorePermissionCheck implementation.");
```

Comment:
to be clear this is not a Jakarta Security 3.0 implementation.  It is a Jakarta EE 1.0 implementation that is also applicable to 2.0 and 3.0 as well.